src/main/kotlin/org/rust/ide/annotator/RsCrateDocLineMarkerProvider.kt

```diff
@@ -0,0 +1,40 @@
+package org.rust.ide.annotator
+
+import com.intellij.codeHighlighting.Pass
+import com.intellij.codeInsight.daemon.LineMarkerInfo
+import com.intellij.codeInsight.daemon.LineMarkerProvider
+import com.intellij.execution.filters.BrowserHyperlinkInfo
+import com.intellij.openapi.editor.markup.GutterIconRenderer
+import com.intellij.psi.PsiElement
+import org.rust.cargo.project.workspace.cargoProject
+import org.rust.ide.icons.RsIcons
+import org.rust.lang.core.psi.RsExternCrateItem
+import org.rust.lang.core.psi.util.module
+
+/**
+ * Provides an external crate imports with gutter icons that open documentation on docs.rs.
+ */
+class RsCrateDocLineMarkerProvider : LineMarkerProvider {
+
+    override fun getLineMarkerInfo(element: PsiElement): LineMarkerInfo<PsiElement>? = null
+
+    override fun collectSlowLineMarkers(elements: MutableList<PsiElement>, result: MutableCollection<LineMarkerInfo<PsiElement>>) {
+        elements.asSequence()
+            .filterIsInstance<RsExternCrateItem>()
+            .forEach { crateItem ->
+                val cargoProject = crateItem.module?.cargoProject ?: return@forEach
+                val (pkg, crate) = cargoProject.findLibTargetByNormName(crateItem.identifier.text)?: return@forEach
+                result.add(LineMarkerInfo(
+                    crateItem.crate,
+                    crateItem.crate.textRange,
+                    RsIcons.DOCS_MARK,
+                    Pass.LINE_MARKERS,
```

Comment:
Can you please check that this doesn't cause duplicate markers under 2016.1 (https://github.com/intellij-rust/intellij-rust/commit/55d34bfe511ddfed9ade998db6d998539fea354f)?